zvariant/src/lib.rs

```diff
@@ -1293,6 +1293,354 @@ mod tests {
         assert_eq!(dict1, dict2);
     }
 
+    fn mock_fd<'a>(value: i32) -> Fd<'a> {
+        use std::os::fd::BorrowedFd;
+        use std::os::fd::RawFd;
+        let fd = unsafe { BorrowedFd::borrow_raw(value as RawFd) };
+        Fd::from(fd)
+    }
+
+    macro_rules! dict_iter_tests_gen {
+        ($($name:ident: ($k_ty:ty, $v_ty:ty, $values:expr),)*) => {
+            $(
+                #[test]
+                fn $name() {
+                    let mut dict = Dict::new(<$k_ty>::signature(), <$v_ty>::signature());
+                    for (k, v) in $values {
+                        dict.add(k, v).unwrap();
+                    }
+                    let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+                    for (k, v) in $values {
+                        expect.push((Value::from(k), Value::from(v)));
+                    }
+                    let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+                    let actual = dict.iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = (&dict).iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = (&mut dict).iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect_iter);
+                    let actual = dict.into_iter().collect::<Vec<_>>();
+                    assert_eq!(actual, expect);
+                }
+            )*
+        }
+    }
+
+    #[test]
+    fn dict_iter_u8_str() {
+        let mut dict = Dict::new(<u8>::signature(), <&str>::signature());
+        let values = vec![(1u8, "123"), (2u8, "456")];
+        for (k, v) in values.iter() {
+            dict.add(k, v).unwrap();
+        }
+        let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+        for (k, v) in values.iter() {
+            expect.push((Value::from(k), Value::from(v)));
+        }
+        let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+        let actual = dict.iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&mut dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = dict.into_iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect);
+    }
+
+    #[test]
+    fn dict_iter_str_u8() {
```

Comment:
* I'd really prefer not to have separate tests for each combo. One test for maps is enough.
* As I previous eluded to, it would actually be great if you can modify the existing test we already have for dict (`dict_value`) and try to add as little as possible.

Reply:
Updated.